apps/meteor/ee/app/livechat-enterprise/server/jobs/AbstractOmniSchedulerClass.ts

```diff
@@ -0,0 +1,44 @@
+import Agenda from 'agenda';
+import { MongoInternals } from 'meteor/mongo';
+import { Collection } from 'mongodb';
+
+import { schedulerLogger } from '../lib/logger';
+
+const SCHEDULER_NAME = 'omnichannel_scheduler';
+
+export abstract class AbstractOmniSchedulerClass {
+	scheduler: Agenda;
+
+	abstract createJobDefinition(): void;
+
+	abstract createIndexes(collection: Collection): void;
+
+	protected constructor() {
+		const { db } = MongoInternals.defaultRemoteCollectionDriver().mongo;
+
+		this.scheduler = new Agenda({
+			mongo: db as any,
+			db: { collection: SCHEDULER_NAME },
+			defaultConcurrency: 1,
+		});
+
+		this.scheduler.on('ready', async () =>
+			this.scheduler.start().then(() => {
+				schedulerLogger.info(`${SCHEDULER_NAME} started`);
+			}),
+		);
+
+		process.on('SIGINT', () => {
+			schedulerLogger.info(`SIGINT received. Stopping ${SCHEDULER_NAME}...`);
+			this.scheduler.stop();
+		});
+		process.on('SIGTERM', () => {
+			schedulerLogger.info(`SIGTERM received. Stopping ${SCHEDULER_NAME}...`);
+			this.scheduler.stop();
+		});
+
+		this.createJobDefinition();
+
+		this.createIndexes(db.collection(SCHEDULER_NAME));
```

Comment:
Can we convert db.collection to a local property? That way if we ever need to extend this, we don't have to change this part :eyes: (by extend i mean: doing other things with the agenda collection)